assignment/final/aLife/code/master.js

```diff
@@ -0,0 +1,417 @@
+                </p>
+                `
+                alreadyAdded.push(makingThisEasier(i));
+            }
+        }
+        alreadyAdded = [];
+    }
+}
+
+function makingThisEasier(i){
+    let subString = pkmnListForTypeFilter[0].LIST[i].pokemon.name.substring(0, pkmnListForTypeFilter[0].LIST[i].pokemon.name.indexOf("-"));
+    //console.log("Substring = " + subString);
+    try{
+        pkmn = pkmnList.filter(pkmn => pkmn.NAME == pkmnListForTypeFilter[0].LIST[i].pokemon.name)[0].NAME;
+        //console.log(pkmn);
+    }catch(err){
+        //console.log("Error from filter on " + pkmnListForTypeFilter[0].LIST[i].pokemon.name);
+        try{
+            pkmn = pkmnList.filter(pkmn => pkmn.NAME == subString)[0].NAME;
+        }catch(err){
+            //console.log("Error from filter on subString");
+            pkmn = "";
+        }
+    }
+    //console.log(pkmn);
+    return pkmn
+}
+
+function removeIds() {
+    for (let i = 0; i < types.length; i++) {
+        if (types[i] != selectedType1 && types[i] != selectedType2) {
+            document.getElementById(types[i]).removeAttribute('id');
+        }
+    }
+}
+
+function replaceIds() {
+    let elements = document.getElementsByClassName("type");
+    for (let i = 0; i < types.length; i++) {
+        elements[i].setAttribute('id', types[i]);
+    }
+}
+
+terminalButton.addEventListener("click", function () {
+    !terminalButton.classList.contains("terminalButtonClicked") ? (
+            addTBClicked(),
+            terminalScreen.innerHTML =
+            `
+            <img src="${currentPkmn.SPRITES.front_shiny}">
+        `) :
+        (removeTBClicked(),
+            terminalScreen.innerHTML =
+            `
+            <img src="${currentPkmn.SPRITES.front_default}">
+        `);
+});
+
+pkmnName.addEventListener("keyup", function () {
+    filterPkmn()
+});
+
+typesDiv.addEventListener("click", function () {
+    removeTBClicked();
+    replaceIds();
+    if (selectedType1 == null) {
+        selectedType1 = event.target.id;
+        //console.log(selectedType1);
+        removeIds();
+    } else {
+        selectedType2 = event.target.id;
+        //console.log(selectedType2);
+        removeIds();
+    }
+    filterByType();
+});
+
+pkmnListDiv.addEventListener("click", function () {
+    removeTBClicked();
+    var x = event.target.id;
+    //console.log(pkmnList.find(pkmn => pkmn.NAME === x));
+    let pkmn = builtPkmn.find(pkmn => pkmn.NAME === x);
+    pkmn != null ? preBuiltPkmn(pkmn) : buildNewPkmn(x);
+});
+
+downId.addEventListener("click", function () {
+    removeTBClicked();
+    let i = pkmnList.indexOf(pkmnList.find(pkmn => pkmn.NAME === currentPkmn.NAME));
+    buildNewPkmn(pkmnList[i - 1].NAME);
+    selectNewPkmn(pkmnList[i - 1].NAME);
+});
+
+upId.addEventListener("click", function () {
+    removeTBClicked();
+    let i = pkmnList.indexOf(pkmnList.find(pkmn => pkmn.NAME === currentPkmn.NAME));
```

Comment:
Same as above for simplifying.

Reply:
I may not know what I’m doing but given enough time I’ll make it work!
Random comment but I really think that fixing this to work properly is going to involve getting the pkmnList from each type instead of the national Pokédex...  May be a more complete list that way And would remove a lot of things that have made filtering overly difficult. But would require me to rewrite everything I think..